### PR TITLE
add unit test for websocket middleware

### PR DESCRIPTION
**Pull request by mstrYoda**
fix: #668

### CODE REVIEW

File: websocket/websocket_test.go [hunk 1]
```diff
@@ -1 +1,60 @@
...
+	"net"
+	"testing"
+
+	"github.com/fasthttp/websocket"
+	"github.com/gofiber/fiber/v2"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestWebSocketMiddleware(t *testing.T) {
+	app := fiber.New()
```
Comment (efectn) [inline websocket/websocket_test.go:13]: Can you disable startup messages

**Review by efectn — COMMENTED** [review]
Can you also add some test cases about websocket config fields and conn methods were added by fiber

File: websocket/websocket_test.go [hunk 2]
```diff
@@ -1 +1,60 @@
...
+				conn.Close()
+				break
+			}
+		}
+	}()
+
+	<-readyCh
+	conn, resp, err := websocket.DefaultDialer.Dial("ws://localhost:3000/ws/message", nil)
+	defer conn.Close()
+	assert.Nil(t, err)
```
Comment (efectn) [inline websocket/websocket_test.go:52]: assert.NoError would be better

**Comment by mstrYoda** [comment]
> Can you also add some test cases about websocket config fields and conn methods were added by fiber

Definetely yes, I will add those too 👍

**Comment by gaby** [comment]
@mstrYoda Once https://github.com/gofiber/contrib/pull/670 this PR will need a few changes, to support the new structure. It will make future updates easier :-)

**Comment by gaby** [comment]
@mstrYoda Can you rebase with master

**Comment by ReneWerner87** [comment]
@mstrYoda @gaby is it ready? can you change the last things from the comments?
would like to merge this

**Comment by gaby** [comment]
> @mstrYoda @gaby is it ready? can you change the last things from the comments? would like to merge this

I think we still waiting on changes

**Comment by mstrYoda** [comment]
> @mstrYoda @gaby is it ready? can you change the last things from the comments? would like to merge this



> > @mstrYoda @gaby is it ready? can you change the last things from the comments? would like to merge this
> 
> I think we still waiting on changes

I will complete it tonight 👍

**Comment by mstrYoda** [comment]
@efectn is it ok for config tests, right now it covers most of the usage 🤔

**Comment by efectn** [comment]
> @efectn is it ok for config tests, right now it covers most of the usage 🤔

Can you add testcases for Locals, Params etc.

Btw what's coverage of the tests ?

**Comment by mstrYoda** [comment]
> > @efectn is it ok for config tests, right now it covers most of the usage 🤔
> 
> Can you add testcases for Locals, Params etc.
> 
> Btw what's coverage of the tests ?

I add all tests for package methods (except joinmessage) coverage is %95.7 right now.

**Review by efectn — APPROVED** [review]

**Review by gaby — APPROVED** [review]
LGTM

**Review by ReneWerner87 — APPROVED** [review]

**Pull request merged**